weave/flow/obj.py

```diff
@@ -63,6 +64,23 @@ def from_uri(cls, uri: str, *, objectify: bool = True) -> Self:
             )
         return api.ref(uri).get(objectify=objectify)
 
+    def publish(self, name: Union[str, None] = None) -> ObjectRef:
+        # I'm using the `publish` term here for now, but ideally we call this `save`.
+        cls_name = self.__class__.__name__
+        if not is_registered(cls_name):
+            raise NotImplementedError("Publish is not supported for this object!")
+
+        return weave_publish(self, name)
+
+    def delete(self) -> None:
+        if self.ref is None:
+            raise ValueError(
+                "Can't delete an object without a ref -- has this object been saved?"
+            )
+
+        self.ref.delete()
+        self.name = f"DELETED OBJECT: {self.name}"
```

Comment:
why change the name? this would break something like
```
obj.publish()
obj.delete()
obj.publish() # new published name is deleted
```

Reply:
good point; fixed

Reply:
Good point.  The intent here is to let the user know that an object they have a handle to has been deleted.  Consider this case:
```py
obj = MyObject(...)
obj.delete()

obj  # looks like it is fine!  How do we warn the user that it's been deleted?
```

Reply:
you could have a deleted flag or something int he repr if you really wanted to. I am not sure why `obj` i bad here. maybe `delete` needs to be more clear that it is `delete_from_remote` or something?

Reply:
Without this, interacting with a deleted obj just feels like a normal obj.  One alternative might be to emit warnings when you interact with the obj, but it could get noisy

Reply:
hmmm, why is `interacting with a deleted obj just feels like a normal obj.` a bad thing? We basically "unpublished" it from the remote, but it is still sitting there in mem

Reply:
hmm, maybe it's not!